internal/service/arcregionswitch/data_source_plan_test.go

```diff
@@ -0,0 +1,266 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccPlanDataSourceConfig_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(dataSourceName, names.AttrARN, resourceName, names.AttrARN),
+					resource.TestCheckResourceAttrPair(dataSourceName, acctest.CtName, resourceName, acctest.CtName),
+					resource.TestCheckResourceAttrPair(dataSourceName, "execution_role", resourceName, "execution_role"),
+					resource.TestCheckResourceAttrPair(dataSourceName, "recovery_approach", resourceName, "recovery_approach"),
+					resource.TestCheckResourceAttrPair(dataSourceName, "regions", resourceName, "regions"),
+					resource.TestCheckResourceAttrPair(dataSourceName, "workflow", resourceName, "workflow"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccARCRegionSwitchPlanDataSource_route53HealthChecks(t *testing.T) {
+	ctx := acctest.Context(t)
+	rName := acctest.RandomWithPrefix(t, acctest.ResourcePrefix)
+	dataSourceName := "data.aws_arcregionswitch_plan.test"
+	resourceName := "aws_arcregionswitch_plan.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(ctx, t)
+			testAccPreCheck(ctx, t)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, "arcregionswitch"),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckPlanDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccPlanDataSourceConfig_route53HealthChecks(rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(dataSourceName, names.AttrARN, resourceName, names.AttrARN),
+					resource.TestCheckResourceAttrPair(dataSourceName, acctest.CtName, resourceName, acctest.CtName),
+					// Verify Route53 health checks API integration works
+					resource.TestCheckResourceAttr(dataSourceName, "route53_health_checks.#", "2"),
+					// Verify health check metadata is immediately available
+					resource.TestCheckResourceAttr(dataSourceName, "route53_health_checks.0.hosted_zone_id", "Z123456789012345678"),
+					resource.TestCheckResourceAttr(dataSourceName, "route53_health_checks.0.record_name", "test.example.com"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "route53_health_checks.0.region"),
+					resource.TestCheckResourceAttr(dataSourceName, "route53_health_checks.1.hosted_zone_id", "Z123456789012345678"),
+					resource.TestCheckResourceAttr(dataSourceName, "route53_health_checks.1.record_name", "test.example.com"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "route53_health_checks.1.region"),
+					// Note: health_check_id fields exist but are empty initially due to AWS 4+ minute delay
+				),
+			},
+		},
+	})
+}
+
+func testAccPlanDataSourceConfig_basic(rName string) string {
+	return fmt.Sprintf(`
+%s
+
+data "aws_arcregionswitch_plan" "test" {
+  arn = aws_arcregionswitch_plan.test.arn
+}
+`, testAccPlanConfig_basic(rName, acctest.AlternateRegion(), acctest.Region()))
+}
```

Comment:
Prefer `acctest.ConfigCompose` to merge multiple Terraform configuration strings together.

Helper functions to generate test configurations should also always go after all acceptance test definitions.